### DISCUSSION TITLE
Offline mode

### DISCUSSION THREAD

**Discussion by WiZuku** [community/community#54670] [Mobile]
Is there any way to use github offline? Like if i want to download something

**Comment by finnjenn** [comment]
Most services on the site require live interaction with their server so I would assume no. You can download a desktop version of github. What do you need to download offline?

**Comment by airtower-luna** [comment]
Depends on *what* you want to use.

Once you clone a *Git* repository, you have an independent local repository (unless you use non-default options to leave out certain things). You can commit, branch, merge, etc. completely offline. When you want to share your changes with others (or receive theirs) you need some kind of connection, but it doesn't have to be via GitHub. I highly recommend you take a look at the [Git Book](https://git-scm.com/book/en/v2) (it's a website, but you can download an ebook, too), chapters 2 through 4 cover the relevant parts.

For GitHub-specific things (like discussions or issues) you need to be online.

**Comment by lotsun** [comment]
what I noticed about using GitHub offline is, it allows you to make the necessary changes, but it doesn't take effect on the repo until you are connected to the internet
